OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -138,8 +139,10 @@ void INotifyCreated.Created(Actor self)
 			// Created is called before Player.PlayerActor is assigned,
 			// so we must query other player traits from self, knowing that
 			// it refers to the same actor as self.Owner.PlayerActor
-			playerPower = (self.Info.Name == "player" ? self : self.Owner.PlayerActor).TraitOrDefault<PowerManager>();
-			productionTraits = self.TraitsImplementing<Production>().ToArray();
+			var playerActor = self.Info.Name == "player" ? self : self.Owner.PlayerActor;
+			playerPower = playerActor.TraitOrDefault<PowerManager>();
+			productionTraits = self.TraitsImplementing<Production>().Where(p => p.Info.Produces.Contains(Info.Type)).ToArray();
```

Comment:
It seems like i'm applying a change from upstream that AS doesn't have yet. But it should break anything, should i revert that?

Reply:
No need, I'll do an engine update immediately after I merge this then.